src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -43,9 +44,350 @@ Copyright (c) 2013, Ford Motor Company
+          }
+        }
+      }
+
+      if (os_type.empty()) {
+        LOG4CXX_WARN(logger_,
+                     kQueryAppsValidationFailedPrefix
+                         << "Can't find mobile OS type in json file.");
+        return false;
+      }
+
+      // Languages verification
+      if (!app_data[os_type].keyExists(json::languages)) {
+        LOG4CXX_WARN(logger_,
+                     kQueryAppsValidationFailedPrefix
+                         << "'languages' doesn't exist");
+        return false;
+      }
+      if (!ValidateLanguages(app_data[os_type][json::languages],
+                             synonyms_map)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  bool ValidateAppIdAndAppName(const smart_objects::SmartObject& app_data,
+                               std::set<std::string>& app_ids_set) const {
+    // Verify appid
+    if (!app_data.keyExists(json::appId)) {
+      LOG4CXX_WARN(logger_,
+                   kQueryAppsValidationFailedPrefix
+                       << "Can't find app ID in json file.");
+      return false;
+    }
+    // Verify appid length
+    const std::string app_id(app_data[json::appId].asString());
+    if (app_id.length() > kAppIdLengthMax) {
+      LOG4CXX_WARN(logger_,
+                   kQueryAppsValidationFailedPrefix
+                       << "An Object ID length exceeds maximum allowed ["
+                       << app_id.length() << "]>[" << kAppIdLengthMax << "]");
+      return false;
+    }
+
+    // Verify that appid is unique
+    if (app_ids_set.find(app_id) != app_ids_set.end()) {
+      LOG4CXX_WARN(logger_,
+                   kQueryAppsValidationFailedPrefix
+                       << "An Object ID is not unigue [" << app_id << "]");
+      return false;
+    }
+    app_ids_set.insert(app_id);
+
+    // Verify that app is not registered yet
+    ApplicationSharedPtr registered_app =
+        manager_.application_by_policy_id(app_id);
+    if (registered_app) {
+      LOG4CXX_WARN(logger_,
+                   kQueryAppsValidationFailedPrefix
+                       << "Application with the same id: " << app_id
+                       << " is registered already.");
+      return false;
+    }
+    // Verify app name exist
+    if (!app_data.keyExists(json::name)) {
+      LOG4CXX_WARN(logger_,
+                   kQueryAppsValidationFailedPrefix
+                       << "Can't find app name in json file.");
+      return false;
+    }
+    // And app name length
+    const std::string appName(app_data[json::name].asString());
+    if (appName.length() > kAppNameLengthMax) {
+      LOG4CXX_WARN(logger_,
+                   kQueryAppsValidationFailedPrefix
+                       << "Name of application exceeds maximum allowed ["
+                       << appName.length() << "]>[" << kAppNameLengthMax
+                       << "].");
+      return false;
+    }
+    return true;
+  }
+
+  bool ValidateLanguages(const smart_objects::SmartObject& languages,
```

Comment:
@okoreniak this method looks too terrible. Is it possible to break it to few separate methods?